### PR TITLE
use REST API with download manager RHD-64

### PR DESCRIPTION
**Pull request by wesbos**
This is almost done - a few things that need to happen

1. The descriptions of the downloads need to be updated in the admin - @Dantheman720  can you help us out here?
2. Logic for when no downloads are available needs to be added - it's there but what to show needs to be added.

### CODE REVIEW

**Comment by paulrobinson** [comment]
Also, you need to update the 'download latest' links on here to use the REST API: http://developers.redhat.com/downloads/

File: javascripts/downloads.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+app.downloads = {};
+
+// app.downloads.url = 'https://developers.stage.redhat.com/download-manager/rest/available/';
+app.downloads.url = 'https://www.stage.jboss.org/download-manager/rest/available/';
```
Comment (LightGuard) [inline javascripts/downloads.js]: Is this the correct production URL? If it isn't we probably need it to be a config variable.

Comment (wesbos) [inline javascripts/downloads.js]: No - it's temporary while we wait for @dhladky to deploy to production

**Comment by wesbos** [comment]
@paulrobinson  Okay - I've checked each page and they all mirror each other after a few small changes. 

The only thing is that the descriptions for each download aren't imported. Should I do this in the admin or do we have someone who can do it?

**Comment by paulrobinson** [comment]
@Dantheman720 can you help @wesbos with importing of the descriptions? @wesbos will be explain in more detail what is needed.

File: javascripts/downloads.js (hunk 1 above)
Comment (paulrobinson) [inline javascripts/downloads.js]: I believe this is in production now?

**Comment by paulrobinson** [comment]
Also, the download data needs to be removed from the product.yml files as we shouldn't be using that anymore.

**Comment by Dantheman720** [comment]
Can one of the admins verify this patch?

**Comment by paulrobinson** [comment]
@Dantheman720 Please can you compare the download pages in this PR to those in Production. You should be able to tweak the config in the DL-manager to make them look the same. Let me know if you have any problems with the DL-man interface.

**Comment by Dantheman720** [comment]
@paulrobinson I am not getting any results from DL-manager.  I think it's because 
jboss_fuse_-_downloads is not pointing to jboss.org/download-manager.  Will this eventually be migrated over?

**Comment by dhladky** [comment]
Hi, 

actually https://developers.redhat.com/download-manager/rest/available/fuse is a valid link for download manager. However if I were you, I would compare your domain and the domain of the DM. I suspect there could be some cross-domain security problems. 

David Hladky

----- Original Message -----

**Comment by paulrobinson** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
@dhladky we need to try this against production in order to ensure the pages will have the same downloads listed. If Dan points this PR to staging dl-man then he will be working with stale data. Plus any fixes he makes will need to be fixed in both staging and production. 

Can you configure the DL-Man to accept requests from it-developers-pr.stage.redhat.com?

**Comment by paulrobinson** [comment]
This seems to be working for me when I visit: http://it-developers-pr.stage.redhat.com/pr/332/build/1295/products/fuse/download/

I see:

https://www.dropbox.com/s/rlbzi5p1phz8gjf/Screenshot%202015-11-24%2019.20.47.png?dl=0

**Comment by paulrobinson** [comment]
@dhladky I'm not seeing the order value in here: https://developers.redhat.com/download-manager/rest/available/fuse. How do we get it?

**Comment by dhladky** [comment]
Hi Paul, 

the order value is not included. It influences the order of the fields, but it is not displayed. Keep in mind, that the order value is not the only thing, that influences the order of the fields. 

David Hladky

----- Original Message -----

**Comment by paulrobinson** [comment]
@dhladky  So how does the front end know what order to display the results?

**Comment by dhladky** [comment]
You need to display it in the same order it is in the rest service. 

David

----- Original Message -----

**Comment by paulrobinson** [comment]
Ah ok, I get it now. Still not sure how to influence the order though. Other than by setting the 'File order' what else can I do?

**Comment by paulrobinson** [comment]
@wesbos: Dan and I are seeing strange issues with the dates.

1. Visit: http://it-developers-pr.stage.redhat.com/pr/332/build/1295/products/fuse/download/
2. I see 6.2.0 released on: '2015-06-01'. Dan sees '2015-06-00'
3. The REST result shows: releaseDate: 1434945600000
4. http://www.epochconverter.com/ thinks 1434945600000 is 'GMT: Mon, 22 Jun 2015 04:00:00 GMT'
5. This is close to http://developers.redhat.com/products/fuse/download/ which shows '2015-06-23'. Discrepancy here is probably due to TZ variations.

**Comment by dhladky** [comment]
The java timestamp is supposed to be independent on the time zone. There could be an issue with how the server time is set up. It would be a question on Vladimir. 

David

----- Original Message -----

**Comment by paulrobinson** [comment]
@wesbos All the middleware download links on here: http://it-developers-pr.stage.redhat.com/pr/332/build/1295/downloads/ go to access.redhat.com. There is also no REST request done on that page.

**Comment by paulrobinson** [comment]
@dhladky I think what you are suggesting explains why it' 1 day out. However the other 21 days is probably die to some JS error.

**Comment by dhladky** [comment]
The "zero" date is definitely a bug in Javascript as the time is in the REST call as a timestamp (long integer) and formated to the page as Javascript.

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
Acceptance tests are failing: http://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/frontend-acceptance-tests/397/console

**Comment by paulrobinson** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@wesbos @paulrobinson tests are good now

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by wesbos** [comment]
@paulrobinson @Dantheman720 Are we good to merge this?

**Comment by paulrobinson** [comment]
retest this please

**Comment by paulrobinson** [comment]
@wesbos hopefully ready. Just waiting for @Dantheman720 to tidy up the config in the download manager so that the download pages look right.

**Comment by paulrobinson** [comment]
@Dantheman720 when this build succeeds, the site will appear at: http://it-developers-pr.stage.redhat.com/pr/332/build/123

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
@wesbos please rebase with master cc @paulrobinson

**Comment by ollyjshaw** [comment]
retest this please

**Comment by wesbos** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by Dantheman720** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
retest this please

**Comment by paulrobinson** [comment]
See: http://it-developers-pr.stage.redhat.com/pr/332/build/186/products/devstudio/download/

Notice that 'Download JBoss Developer Studio from the Red Hat Customer Portal' link is present, despite there being download options visible.

**Comment by paulrobinson** [comment]
Also, JS doesn't seem to be firing on http://it-developers-pr.stage.redhat.com/pr/332/build/186/downloads/

**Comment by paulrobinson** [comment]
I'm also seeing the data problem with Fuse that I mentioned above.

**Comment by paulrobinson** [comment]
The EAP Download page is not looking right. Compare

Live: http://developers.redhat.com/products/eap/download/
PR: http://it-developers-pr.stage.redhat.com/pr/332/build/195/products/eap/download/

Notice that when a non-GA is available, its not showing the latest GA as well. 

Also, don't forget to make sure that these links are correct too: https://www.dropbox.com/s/qplsotutyoh103q/Screenshot%202015-12-07%2010.46.02.png?dl=0

**Comment by paulrobinson** [comment]
Also, there are now two 'Zip Files' entries for EAP 6.4. One of them needs to be hidden as the second is a replacement. I'll need to speak to @dhladky about how to do this.

**Comment by wesbos** [comment]
That is because we only have one "featured artifact" coming back from the DCP 

https://developers.redhat.com/download-manager/rest/available/eap

**Comment by paulrobinson** [comment]
Looks like a rebase is needed.

**Comment by paulrobinson** [comment]
When multiple versions are present on the page, the table is combined. 

See: http://it-developers-pr.stage.redhat.com/pr/332/build/225/products/eap/download/

It should look like: http://developers.redhat.com/products/eap/download/

**Comment by paulrobinson** [comment]
See: http://it-developers-pr.stage.redhat.com/pr/332/build/225/products/brms/download/. Notice the full product name is used (Download JBoss Business Rule Management System 6.2.0), rather than the abbreviated code used on: http://developers.redhat.com/products/brms/download/

**Comment by wesbos** [comment]
@paulrobinson Okay - I've put each in their own table. Each product doesn't have a featured download so we only have the title for the top featured item

**Comment by paulrobinson** [comment]
Please can you rebase?

**Comment by wesbos** [comment]
@paulrobinson done

**Comment by paulrobinson** [comment]
cheers!

**Pull request merged**